### PR TITLE
Fix CID 78557

### PR DESCRIPTION
**Pull request by milabs**

### CODE REVIEW

File: tempesta_fw/http_msg.c [hunk 1]
```diff
@@ -841,6 +841,12 @@ tfw_http_msg_alloc(int type)
 	ss_skb_queue_head_init(&hm->msg.skb_list);
 
 	hm->h_tbl = (TfwHttpHdrTbl *)tfw_pool_alloc(hm->pool, TFW_HHTBL_SZ(1));
+	if (!hm->h_tbl) {
```
Comment (keshonok) [inline tempesta_fw/http_msg.c]: Perhaps, s simpler and faster `tfw_pool_destroy(m->pool);` would suffice here?
Also, perhaps the statement above `ss_skb_queue_head_init(&hm->msg.skb_list);` should better be moved lower and reside after the memory allocation.

We don't have a message yet that is known to the outside of this little function. But calling `tfw_http_msg_free(hm);` would tell that we're freeing a known message (that really doesn't exist yet in all meaningful terms). Extra generic actions would be executed on the message. And a debug message would be printed about freeing a message that didn't appear anywhere yet.

Comment (milabs) [inline tempesta_fw/http_msg.c]: Ok, but what's about `hm` itself?

Comment (milabs) [inline tempesta_fw/http_msg.c]: Ok, I got it.

File: tempesta_fw/http_msg.c [hunk 2]
```diff
@@ -838,13 +838,18 @@ tfw_http_msg_alloc(int type)
 	if (!hm)
 		return NULL;
 
-	ss_skb_queue_head_init(&hm->msg.skb_list);
-
 	hm->h_tbl = (TfwHttpHdrTbl *)tfw_pool_alloc(hm->pool, TFW_HHTBL_SZ(1));
+	if (!hm->h_tbl) {
```
Comment (krizhanovsky) [inline tempesta_fw/http_msg.c]: please use `unlikely()`

File: tempesta_fw/http_msg.c [hunk 3]
```diff
@@ -838,13 +838,18 @@ tfw_http_msg_alloc(int type)
...
+		TFW_WARN("Insufficient memory to create message\n");
+		tfw_pool_destroy(m->pool);
```
Comment (krizhanovsky) [inline tempesta_fw/http_msg.c]: must be `hm`, not `m`.

File: tempesta_fw/http_msg.c [hunk 4]
```diff
@@ -838,13 +838,18 @@ tfw_http_msg_alloc(int type)
 	if (!hm)
```
Comment (krizhanovsky) [inline tempesta_fw/http_msg.c]: Please put `unlikely()` here as well.

**Comment by krizhanovsky** [comment]
good to merge with the fixe and cleanups

**Pull request merged**